### PR TITLE
Use getClassLoader() instead of Thread.currentThread().getContextClas…

### LINKED ISSUE #1272: Don't use thread context's class loader for Android.
As http://stackoverflow.com/questions/13407006/android-class-loader-may-fail-for-processes-that-host-multiple-applications says, if users use android:sharedUserId or android:process, the thread context's class loader will be replaced with a dummy implementation.

And calling Thread.currentThread().setContextClassLoader() before whenever the ManagedChannelBuilder is created is silly.

After discussed with @ejona86, we agreed to use this.getClass().getClassLoader() for Android case.

### PR DESCRIPTION
**Pull request by madongfly**
…sLoader() for android.

See #1272 for more details.

Fixes #1272.

@ejona86 Please take a look, thanks!

### CODE REVIEW

**Comment by ejona86** [comment]
@madongfly LGTM

File: core/src/main/java/io/grpc/ManagedChannelProvider.java [hunk 1]
```diff
@@ -87,6 +87,27 @@ public static ManagedChannelProvider provider() {
     return provider;
   }
 
+  private static ClassLoader getCorrectClassLoader() {
+    if (isAndroid()) {
+      // When android:sharedUserId or android:process is used, Android will setup a dummy
+      // ClassLoader for the thread context (http://stackoverflow.com/questions/13407006),
+      // instead of letting users to manually set context class loader, we choose the
+      // correct class loader here.
+      return ManagedChannelProvider.class.getClassLoader();
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/ManagedChannelProvider.java]: Will the dummy class loader be able to read from the META-INF file?

Also, are there any issues with the Class.forName below?  Doesn't that use the current class loader, which in this case would be the dummy one?

Comment (ejona86) [inline core/src/main/java/io/grpc/ManagedChannelProvider.java]: `Class.forName` would use `ManagedChannelProvider.class.getClassLoader()`, to my understanding.

There are some concerns about using the write class loader _in general_, but on Android @madongfly was saying that all classes in the same APK are in the same class loader, which is what I would have expected.

Comment (madongfly) [inline core/src/main/java/io/grpc/ManagedChannelProvider.java]: @carl-mastrangelo, no, the dummy class loader just output a warning to logcat and will not load anything.

Yes, as [Class.forName](http://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#forName(java.lang.String)) says:
```
Returns the Class object associated with the class or interface with the given string name. Invoking this method is equivalent to:
Class.forName(className, true, currentLoader)
where currentLoader denotes the defining class loader of the current class.
```

**Pull request merged**